.github/workflows/ci-test-demos.yaml

```diff
@@ -0,0 +1,58 @@
+name: ci-test-demos
+
+on:
+  push:
+    branches:
+      - main
+      - stable/*
+      - rc/*
+  pull_request:
+    branches:
+      - main
+      - stable/*
+      - rc/*
+
+jobs:
+  test-demo-starter-go:
+    name: test-demo-starter-go
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./clients/go
+    services:
+      sapphire-localnet-ci:
+        image: ghcr.io/oasisprotocol/sapphire-localnet:latest
+        ports:
+          - 8545:8545
+          - 8546:8546
+        env:
+          OASIS_DOCKER_START_EXPLORER: no
+        options: >-
+          --rm
+          --health-cmd="test -f /CONTAINER_READY"
+          --health-start-period=90s
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.x"
+
+      - uses: actions/checkout@v4
+        with:
+          repository: oasisprotocol/demo-starter-go
+          path: ./clients/go/demo-starter-go
```

Comment:
Rather use an external path like `/demos/demo-starter-go` just to make sure there are no potential issues with go sub-packages.